lib/net.c

```diff
@@ -1547,151 +1545,79 @@ start_iface(struct gatekeeper_if *iface, unsigned int num_attempts_link_get)
 	return ret;
 }
 
-static void
-calculate_net_config_para(struct net_config *net_conf)
+unsigned int
+calculate_mempool_config_para(const char *block_name,
+	struct net_config *net_conf, unsigned int total_pkt_burst)
 {
-	int i;
-
-	const char *filtered_blocks[] = { "dynamic_conf" };
-
-	/*
-	 * The total number of lcores used by any functional block
-	 * for either RX or TX except for Dynamic Configuration.
-	 */
-	int gatekeeper_num_lcores = launch_count_lcores(filtered_blocks,
-		RTE_DIM(filtered_blocks));
+	unsigned int num_mbuf;
 
 	/*
 	 * The total number of receive descriptors to
-	 * allocate for the receive ring over all interfaces.
+	 * allocate per lcore for the receive ring over all interfaces.
 	 */
-	uint16_t gatekeeper_total_rx_desc = net_conf->front.num_rx_queues *
-		net_conf->front.num_rx_desc + (net_conf->back_iface_enabled ?
-		net_conf->back.num_rx_queues * net_conf->back.num_rx_desc : 0);
+	uint16_t total_rx_desc = net_conf->front.num_rx_desc +
+		(net_conf->back_iface_enabled ? net_conf->back.num_rx_desc : 0);
 
 	/*
 	 * The total number of transmit descriptors to
-	 * allocate for the transmit ring over all interfaces.
+	 * allocate per lcore for the transmit ring over all interfaces.
 	 */
-	uint16_t gatekeeper_total_tx_desc = net_conf->front.num_tx_queues *
-		net_conf->front.num_tx_desc + (net_conf->back_iface_enabled ?
-		net_conf->back.num_tx_queues * net_conf->back.num_tx_desc : 0);
+	uint16_t total_tx_desc = net_conf->front.num_tx_desc +
+		(net_conf->back_iface_enabled ? net_conf->back.num_tx_desc : 0);
 
 	/*
 	 * The number of elements in the mbuf pool.
 	 *
 	 * Need to provision enough memory for the worst case.
-	 * It's the number of RX descriptors (across all queues for all ports),
-	 * the number of TX descriptors (across all queues for all ports),
-	 * the number of packet burst buffers (across all lcores and
-	 * all interfaces including the KNI), the number of slots in
-	 * the packet buffer cache (across all lcores).
+	 * It's the number of RX descriptors, the number of TX descriptors,
+	 * the number of packet burst buffers (all interfaces
+	 * including the KNI), the number of slots in the packet buffer cache.
 	 */
-	uint32_t gatekeeper_max_num_pkt = gatekeeper_total_rx_desc +
-		gatekeeper_total_tx_desc + net_conf->front.total_pkt_burst +
-		net_conf->back.total_pkt_burst + gatekeeper_num_lcores *
-		RTE_MEMPOOL_CACHE_MAX_SIZE;
+	uint32_t max_num_pkt = total_rx_desc + total_tx_desc +
+		total_pkt_burst + RTE_MEMPOOL_CACHE_MAX_SIZE;
```

Comment:
Given that we are not using the cache, can't we drop `RTE_MEMPOOL_CACHE_MAX_SIZE`?